internal/kuberesource/mutators_test.go

```diff
@@ -416,3 +430,27 @@ func TestAddServiceMesh(t *testing.T) {
 		})
 	}
 }
+
+func TestAddImageStore_Regression(t *testing.T) {
+	for name, tc := range map[string]struct {
+		resource []byte
+	}{
+		"job":        {resource: job},
+		"deployment": {resource: deployment},
+		"cronjob":    {resource: cronjob},
+		"daemonset":  {resource: daemonset},
+		"pod":        {resource: pod},
+	} {
+		t.Run(name, func(t *testing.T) {
+			require := require.New(t)
+
+			res, err := UnmarshalApplyConfigurations(tc.resource)
+			require.NoError(err)
+			res = PatchRuntimeHandlers(res, "contrast-cc")
+			res = AddImageStore(res)
+			encoded, err := EncodeResources(res...)
```

Comment:
Am I correct in assuming the encoding is done because converting the `[]any` to the different resource types is a hassle?

Reply:
Exactly. I don't mean for this to be an exhaustive test (we should still add that), but rather a smoke detector.

Reply:
Makes sense!